### ISSUE TITLE
authentication

### ISSUE THREAD

**Issue by essadji** [ttutisani/CouchDBClient#5]
Any plans to make that possible?

**Comment by ttutisani** [comment]
Based on CouchDB online documentation (http://docs.couchdb.org/en/2.1.1/intro/security.html?highlight=basic%20authentication - section "Basic Authentication"), you can easily achieve authentication by putting username/password in the server url when making requests.

e.g. `http://anna:secret@127.0.0.1:5984` - where "anna" is a username, and "secret" is a password. "127.0.0.1:5984" is the running instance of the CouchDB.

So, you can pass this url to the `CouchDBServer` constructor and that should do it.
Does that solve your problem?

**Comment by essadji** [comment]
Not really. I had already found those instructions, of course. In the case which was handed to me, however, the password I have to use contains periods and a hash. Unfortunately the above method doensn't react well to those characters. Hence my question. (did not find a way to escape problematic chars either, by the way)

**Comment by ttutisani** [comment]
I think you need to url-encode the password before putting it into the server url. If it contains hash that will be replaced after encoding and not a problem. But unfortunately period does not change after url-encoding, so maybe you should use a different password, if you can.

I may look into implementing authentication besides basic one, but that will take some time (maybe couple months?). 

If you can't wait, implement it in your fork and pull request back.
If unsure about the best approach, I can take on drafting UML diagram for it. Designing it right in advance is usually fast (couple of days). Coding is what takes time.

**Comment by essadji** [comment]
Hello again,

I've put this aside for a while but...
Still can't log in with the software.
Already added an admin account without special chars but I always get 'unauthorized' over https
Over http, I get an unecpected char error (apparently the JSON parser gets a < as first input char?)
On another sever, I do get access over http (not https on this one) but it has no PW set

Does that make sense to you or trigger some intuitive solution?

BTW: hashtag does not parse in a url, I've had to encode that too

**Comment by ttutisani** [comment]
That reminds me issues that I had with CouchDB local instance. I couldn't get it authenticate my requests even using regular Postman or other http clients. That was the reason I stopped implementing authentication, because I was not sure how it would work from inside the framework if it didn't work without it.

So, if you can describe how authentication works without CouchDBClient, at least that can serve as a requirement how it should work from CouchDBClient too. In other words, did you get it working without CouchDBClient at all?

**Comment by essadji** [comment]
Yes, I did. But not from C# code. Would JS-code be helpful?

________________________________
Van: Tengiz Tutisani <notifications@github.com>
Verzonden: maandag 16 april 2018 15:32:56
Aan: ttutisani/CouchDBClient
CC: Siegmund Leducq; Author
Onderwerp: Re: [ttutisani/CouchDBClient] authentication (#5)


That reminds me issues that I had with CouchDB local instance. I couldn't get it authenticate my requests even using regular Postman or other http clients. That was the reason I stopped implementing authentication, because I was not sure how it would work from inside the framework if it didn't work without it.

So, if you can describe how authentication works without CouchDBClient, at least that can serve as a requirement how it should work from CouchDBClient too. In other words, did you get it working without CouchDBClient at all?

**Comment by ttutisani** [comment]
Yes, JS is also fine.
